controllers/medusa/medusabackupschedule_controller.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package medusa
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	medusav1alpha1 "github.com/k8ssandra/k8ssandra-operator/apis/medusa/v1alpha1"
+	cron "github.com/robfig/cron/v3"
+)
+
+type Clock interface {
+	Now() time.Time
+}
+
+// MedusaBackupScheduleReconciler reconciles a MedusaBackupSchedule object
+type MedusaBackupScheduleReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Clock  Clock
+}
+
+type RealClock struct{}
+
+func (r *RealClock) Now() time.Time {
+	return time.Now()
+}
+
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules/finalizers,verbs=update
+// +kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=cassandrabackups,verbs=get;list;watch;create
+
+func (r *MedusaBackupScheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("medusabackupscheduler", req.NamespacedName)
+
+	backupSchedule := &medusav1alpha1.MedusaBackupSchedule{}
+	err := r.Get(ctx, req.NamespacedName, backupSchedule)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	sched, err := cron.ParseStandard(backupSchedule.Spec.CronSchedule)
+	if err != nil {
+		// The schedule is in incorrect format
+		return ctrl.Result{}, err
+	}
+
+	previousExecution, err := getPreviousExecutionTime(ctx, backupSchedule)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	now := r.Clock.Now().UTC()
+
+	// Calculate the next execution time
+	nextExecution := sched.Next(previousExecution).UTC()
+
+	createBackup := false
+
+	if nextExecution.Before(now) {
```

Comment:
How hard do you think it would be to check if the previous execution is still running here as well, to prevent starting multiple backups concurrently?
That's something Medusa is missing, sadly.